jablib/src/main/java/org/jabref/model/database/BibDatabaseContext.java

```diff
@@ -282,17 +285,30 @@ public Path getFulltextIndexPath() {
         return indexPath;
     }
 
-    public static BibDatabaseContext of(String bibContent, ImportFormatPreferences importFormatPreferences) throws JabRefException {
+    public static BibDatabaseContext of(Reader bibContentReader, ImportFormatPreferences importFormatPreferences) throws JabRefException {
         BibtexParser parser = new BibtexParser(importFormatPreferences);
         try {
-            Reader reader = Reader.of(bibContent);
-            ParserResult result = parser.parse(reader);
+            ParserResult result = parser.parse(bibContentReader);
             return result.getDatabaseContext();
         } catch (IOException e) {
             throw new JabRefException("Failed to parse BibTeX content", e);
         }
     }
 
+    public static BibDatabaseContext of(String bibContent, ImportFormatPreferences importFormatPreferences) throws JabRefException {
+        return of(Reader.of(bibContent), importFormatPreferences);
+    }
+
+    public static BibDatabaseContext of(InputStream bibContentStream, ImportFormatPreferences importFormatPreferences) throws JabRefException {
+        try {
+            try (Reader reader = new BufferedReader(new InputStreamReader(bibContentStream))) {
+                return of(reader, importFormatPreferences);
+            }
+        } catch (IOException e) {
```

Comment:
I think we can collapse these
```suggestion
        try (Reader reader = new BufferedReader(new InputStreamReader(bibContentStream))) {
            return of(reader, importFormatPreferences);
        } catch (IOException e) {
```